src/substrait/textplan/StructuredSymbolData.h

```diff
@@ -6,23 +6,31 @@
 #include <optional>
 
 #include "substrait/proto/algebra.pb.h"
+#include "substrait/textplan/Location.h"
 
 namespace io::substrait::textplan {
 
+class SymbolInfo;
+
 // Used by the PlanRelation and Relation to track connectivity.
 struct RelationData {
-  RelationData(
-      const ::substrait::proto::Rel* relation,
-      const ::google::protobuf::Message* origin)
-      : protoAddr(relation), originProtoAddr(origin) {
-    continuingPipeline = nullptr;
-  };
-
-  const ::substrait::proto::Rel* protoAddr;
-  const ::google::protobuf::Message* originProtoAddr;
-
-  const ::substrait::proto::Rel* continuingPipeline;
-  std::vector<const ::substrait::proto::Rel*> newPipelines;
+  explicit RelationData(const Location location) : symbolLocation(location){};
+
+  const Location symbolLocation;
+
+  // For relations not starting a pipeline, this is the relation that begins the
+  // pipeline that this relation is part of.
+  const SymbolInfo* pipelineStart{nullptr};
+
+  const SymbolInfo* continuingPipeline{nullptr};
```

Comment:
This will result in an error since c will be starting multiple pipelines.  But for the first two lines the results will be:

```
       pipelineStart   continuingPipeline   newPipelines
a.    c                       nullptr                        {}
b.    c                       a                                 {}
c.    d                       nullptr                        {b}
d.    kUnknown       nullptr                        {d}
```